boards/esp32c3.json

```diff
@@ -5,7 +5,7 @@
       "memory_type": "dio_qspi"
     },
     "core": "esp32",
-    "extra_flags": "-DARDUINO_USB_CDC_ON_BOOT=0 -DESP32_4M -DESP32C3 -DCONFIG_IDF_TARGET_ESP32C3=1",
+    "extra_flags": "-DARDUINO_USB_CDC_ON_BOOT=0 -DESP32_4M -DESP32C3 -DESP_IDF_VERSION_MAJOR=4 -DCONFIG_IDF_TARGET_ESP32C3=1",
```

Comment:
Not sure we should have the IDF major version in the board definition.
Makes more sense to have it in the PIO env where we also define the platform.

Reply:
Moved